test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -667,6 +667,35 @@ TEST_F(HttpConnectionManagerConfigTest, UnixSocketInternalAddress) {
   EXPECT_FALSE(config.internalAddressConfig().isInternalAddress(externalIpAddress));
 }
 
+TEST_F(HttpConnectionManagerConfigTest, CidrRangeBasedInternalAddress) {
+  const std::string yaml_string = R"EOF(
+  stat_prefix: ingress_http
+  internal_address_config:
+    cidr_ranges:
+    - address_prefix: 100.64.0.0
+      prefix_len: 10
+    - address_prefix: 50.20.0.0
+      prefix_len: 20
+  route_config:
+    name: local_route
+  http_filters:
+  - name: envoy.filters.http.router
+  )EOF";
+
+  HttpConnectionManagerConfig config(parseHttpConnectionManagerFromYaml(yaml_string), context_,
+                                     date_provider_, route_config_provider_manager_,
+                                     scoped_routes_config_provider_manager_, http_tracer_manager_,
+                                     filter_config_provider_manager_);
+  Network::Address::Ipv4Instance firstInternalIpAddress{"100.64.0.10", 0, nullptr};
+  Network::Address::Ipv4Instance secondInternalIpAddress{"50.20.0.5", 0, nullptr};
+  Network::Address::Ipv4Instance defaultIpAddress{"10.48.179.130", 0, nullptr};
```

Comment:
Please add a comment explaining that this address is in the list of acceptable addresses when the new config is unset, and that the test is verifying that it doesn't match now because the config is set/non-empty.